rust/src/dcerpc/dcerpc.rs

```diff
@@ -1177,14 +1177,22 @@ pub extern "C" fn rs_dcerpc_parse_request(
     }
     if input_len > 0 && input != std::ptr::null_mut() {
         let buf = build_slice!(input, input_len as usize);
-        return state.handle_input_data(buf, core::STREAM_TOSERVER);
+        let ret = state.handle_input_data(buf, core::STREAM_TOSERVER);
+        let tx_id = state.tx_id - 1;
+        let req_done = if let Some(tx) = state.get_tx(tx_id) {
```

Comment:
for the record: if this works it needs to be cleaned up to avoid the tx lookup

Reply:
OK. On it.